### ISSUE TITLE
[Bug]: Continuous modification of user information results in an error

### ISSUE THREAD

**Issue by niyaco** [gothinkster/realworld#1374]
### Relevant scope

Deployed demo

### Description

When I try to modify user information for the second time consecutively, the server returns a status code of 500 and displays the following error message: “\nInvalid prisma.user.update() invocation:\n\n\nAn operation failed because it depends on one or more records that were required but not found. Record to update not found.”

<img width="1426" alt="image" src="https://github.com/gothinkster/realworld/assets/53268471/4864beb3-b652-46ec-bf94-12d4a6e2829b">

**Comment by geromegrignon** [comment]
Hi @niyaco,

I can reproduce it.
I'm starting local tests right now to identify the bug.

**Comment by geromegrignon** [comment]
I'm quite busy right now handling it but based on my first tests, it's due to using the username from the token on backend side to find the user to be updated.

It happens by firstly updating the username.
On consequential tries, you still have the same token with the old username.
The API consumes it to find the user to be updated, but there is nothing to be found.

**Comment by rahulthedevil** [comment]
> I'm quite busy right now handling it but based on my first tests, it's due to using the username from the token on backend side to find the user to be updated.
> 
> It happens by firstly updating the username. On consequential tries, you still have the same token with the old username. The API consumes it to find the user to be updated, but there is nothing to be found.

Then why not use something that does not change to generate the token? Maybe user id? If it seems that it may cause security issue, then why not create one unique id for each user only to use it for access token and it will never change and eliminate all data from token which can be changed by user

**Comment by geromegrignon** [comment]
> > I'm quite busy right now handling it but based on my first tests, it's due to using the username from the token on backend side to find the user to be updated.
> > It happens by firstly updating the username. On consequential tries, you still have the same token with the old username. The API consumes it to find the user to be updated, but there is nothing to be found.
> 
> Then why not use something that does not change to generate the token? Maybe user id? If it seems that it may cause security issue, then why not create one unique id for each user only to use it for access token and it will never change and eliminate all data from token which can be changed by user

Yes exactly, that's what I have in mind (there is already a unique id for each user, but not exposed to the API currently).
